src/main/java/org/opensearch/geospatial/action/upload/geojson/UploadGeoJSONRequestContent.java

```diff
@@ -61,9 +64,28 @@ public static UploadGeoJSONRequestContent create(Map<String, Object> input) {
                 geoJSONData + " is not an instance of List, but of type [ " + geoJSONData.getClass().getName() + " ]"
             );
         }
+        final long featureCount = getFeatureCount(geoJSONData);
+        if (featureCount > MAX_SUPPORTED_GEOJSON_FEATURE_COUNT) {
```

Comment:
Also, can we break this down into a separate method like `validateFeatureCount(geoJSONData) { ....}` ?

Reply:
created new method